File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findDuplicate = function(nums) {
    let slow = 0;
    let fast = 0;
    while (true) {
        slow = nums[slow];
        fast = nums[fast];
        fast = nums[fast];;
        if (slow === fast) {
            slow = 0;
            while (slow !== fast) {
                slow = nums[slow];
                fast = nums[fast];
            }
            return slow;
        }
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.go
func findDuplicate(nums []int) int {
    slow := 0;
    fast := 0;
    for true {
        slow = nums[slow];
        fast = nums[fast];
        fast = nums[fast];
        if slow == fast {
            slow = 0;
            for slow != fast {
                slow = nums[slow];
                fast = nums[fast];
            }
            return slow;
        }
    }
    return 0;
}